src/resolve/resolved-dns-packet.c

```diff
@@ -1927,6 +1936,10 @@ int dns_packet_read_rr(DnsPacket *p, DnsResourceRecord **ret, bool *ret_cache_fl
                 if (r < 0)
                         return r;
 
+                if ((offset + rdlength - p->rindex <          rdlength - p->rindex) ||
+                    (offset + rdlength - p->rindex > offset + rdlength))
+                        return -EOVERFLOW;
```

Comment:
And isn't this equivalent to `rdlength < p->rindex - rdlength`?

Reply:
You mean rdlength + offset < p->rindex, and that is the bug causing the tests to fail.